compio-quic/benches/quic.rs

```diff
@@ -0,0 +1,196 @@
+                        async {
+                            send.write_all(content).await.unwrap();
+                            send.finish().unwrap();
+                        },
+                        async {
+                            let mut buf = vec![];
+                            recv.read_to_end(&mut buf).await.unwrap();
+                        }
+                    );
+                })
+                .collect::<FuturesUnordered<_>>();
+            while futures.next().await.is_some() {}
+        }
+        drop(handle);
+        start.elapsed()
+    })
+}
+
+fn echo_quinn(b: &mut Bencher, content: &[u8], streams: usize) {
+    use quinn::{ClientConfig, Endpoint, ServerConfig};
+
+    let runtime = tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()
+        .unwrap();
+    b.to_async(&runtime).iter_custom(|iter| async move {
+        let (cert, key_der) = gen_cert();
+        let server_config = ServerConfig::with_single_cert(vec![cert.clone()], key_der).unwrap();
+        let mut roots = rustls::RootCertStore::empty();
+        roots.add(cert).unwrap();
+        let client_config = ClientConfig::with_root_certificates(Arc::new(roots)).unwrap();
+        let server = Endpoint::server(
+            server_config,
+            SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), 0),
+        )
+        .unwrap();
+        let mut client =
+            Endpoint::client(SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), 0)).unwrap();
+        client.set_default_client_config(client_config);
+        let addr = server.local_addr().unwrap();
+
+        let (client_conn, server_conn) = futures_util::join!(
+            async move { client.connect(addr, "localhost").unwrap().await.unwrap() },
+            async move { server.accept().await.unwrap().await.unwrap() }
+        );
+
+        let start = Instant::now();
+        tokio::spawn(async move {
+            while let Ok((mut send, mut recv)) = server_conn.accept_bi().await {
+                tokio::spawn(async move {
+                    echo_impl!(send, recv);
+                });
+            }
+        });
+        for _i in 0..iter {
+            let mut futures = (0..streams)
+                .map(|_| async {
+                    let (mut send, mut recv) = client_conn.open_bi().await.unwrap();
+                    tokio::join!(
+                        async {
+                            send.write_all(content).await.unwrap();
+                            send.finish().unwrap();
+                        },
+                        async {
+                            recv.read_to_end(usize::MAX).await.unwrap();
+                        }
+                    );
+                })
+                .collect::<FuturesUnordered<_>>();
+            while futures.next().await.is_some() {}
+        }
+        start.elapsed()
+    });
+}
+
+fn echo(c: &mut Criterion) {
+    let mut rng = thread_rng();
+
+    let mut large_data = [0u8; 1024 * 64];
```

Comment:
Use `Vec`, and you can use a larger array.

Reply:
Oh I forgot 🤣